test/unit/org/apache/cassandra/index/sai/disk/v1/LegacyOnDiskFormatTest.java

```diff
@@ -158,6 +159,7 @@ public void canCreateAndUsePrimaryKeyMapWithLegacyFormat() throws Throwable
         assertEquals(countFromFactory, countFromMap);
 
         PrimaryKey expected = pkFactory.createTokenOnly(Murmur3Partitioner.instance.decorateKey(Int32Type.instance.decompose(23)).getToken());
+        assertTrue(expected.isTokenOnly());
```

Comment:
let's add assertions in other existing tests (and/or add new tests?) where we expect that isTokenOnly() returns false